### PR TITLE
Virtual Files: Allow to download a folder recursively from the socketAPI

### PR DESCRIPTION
**Pull request by ogoffart**
Issue: #6466

### CODE REVIEW

**Comment by ogoffart** [comment]
(I think this could still go in 2.5...  otherwise we can just re-target)

**Comment by ogoffart** [comment]
This allow to add the "Download files" menu entry for a folder in a sync directory with virtual files. 
This will download all the virtual files.
Note that future files added in that folder on the server will still be virtual files.

cc @hodyroff

File: src/gui/socketapi.cpp [hunk 1]
```diff
@@ -757,11 +757,11 @@ void SocketApi::command_GET_MENU_ITEMS(const QString &argument, OCC::SocketListe
     }
 
     // Virtual file download action
-    if (folder) {
+    if (folder && folder->useVirtualFiles()) {
```
Comment (ckamm) [inline src/gui/socketapi.cpp]: Hm, we might want to rename/document this better. Actually, this setting just controls whether new files will start out as virtual or not. It's perfectly possible to switch this of (so all new files are no longer virtual) but still have heaps of virtual files in the sync directories.

Short: I'd show this option even for folders that don't have the option enabled.

**Comment by ckamm** [comment]
I'm fine with this going into 2.5 since it just adds a new option to a feature that's experimental in the first place

File: src/gui/socketapi.cpp (hunk 1 above)
Comment (ogoffart) [reply src/gui/socketapi.cpp]: Right, in the future, we could decouple the two.
The problem is that I don't really want to recursively go into the folder to see if there are placeholders.
(altough maybe a SELECT WHERE type=4 AND IS_PREFIX_PATH_OF might be simple enough)

Comment (ckamm) [reply src/gui/socketapi.cpp]: What about moving the check down to the files loop then and only removing the action from directories when the folder setting is disabled? Then people could at least still download files individually if the option is off.

**Pull request merged**